app/controllers/users_controller.rb

```diff
@@ -0,0 +1,10 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!
+
+  def downgrade
```

Comment:
You could move this logic into the model on user and do a unit test. You would test that the role changes to standard when it was premium before. Hard to test on the controller level and writing controller level tests for this stuff is no fun!